node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -349,11 +349,14 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 						#channel_name_rx, #channel_name_unbounded_rx
 					);
 					let (signal_tx, signal_rx) = #support_crate ::metered::channel(SIGNAL_CHANNEL_CAPACITY);
+					let subsystem_string = stringify!(#subsystem_name);
```

Comment:
One idea is to have an annotation, since right now the string is just the Overseer struct field name. We can then use it across the codebase. I'm open to any other ideas.

Reply:
I don't think that's necessary, re-use is good and keeps things consistent. I said `derived` with transformations such as `snake_case`ing i.e. in mind for consistency once that is agreed upon / unified.